components/search/SearchInterface.vue

```diff
@@ -350,16 +351,17 @@
       rerouteSearch(queryUpdates) {
         return this.$router.push(this.localePath({ ...this.route, ...{ query: this.updateCurrentSearchQuery(queryUpdates) } }));
       },
-      queryUpdatesForFacetChanges(selected) {
+      queryUpdatesForFacetChanges(selected = {}) {
         const filters = Object.assign({}, this.filters);
 
         for (const name in selected) {
           filters[name] = selected[name];
+        }
 
-          // Remove collection-specific filters when collection is changed
-          // TODO: this is rather crude and will not scale well; improve
-          if (name === 'THEME' && !selected['api']) {
-            filters['api'] = null;
+        // Remove collection-specific filters when collection is changed
+        if (selected[this.THEME]) {
```

Comment:
This isn't catching the case where the theme is reset. Meaning you can end up on a page like this:
http://localhost:3000/en/search?page=1&qf=proxy_dc_format.en%3A%22Technique%3A%20embroidery%22&query=&view=grid